backend/api/Controllers/RobotController.cs

```diff
@@ -643,34 +643,41 @@ public async Task<ActionResult> ResumeMission([FromRoute] string robotId)
     /// </remarks>
     [HttpPost]
     [Authorize(Roles = Role.User)]
-    [Route("{robotId}/start-localization")]
+    [Route("start-localization")]
     [ProducesResponseType(typeof(Mission), StatusCodes.Status200OK)]
     [ProducesResponseType(StatusCodes.Status400BadRequest)]
     [ProducesResponseType(StatusCodes.Status401Unauthorized)]
     [ProducesResponseType(StatusCodes.Status403Forbidden)]
     [ProducesResponseType(StatusCodes.Status404NotFound)]
     [ProducesResponseType(StatusCodes.Status500InternalServerError)]
     public async Task<ActionResult<Mission>> StartLocalizationMission(
-        [FromRoute] string robotId,
-        [FromBody] Pose localizationPose
+        [FromBody] ScheduleLocalizationMissionQuery scheduleLocalizationMissionQuery
+
     )
     {
-        var robot = await _robotService.ReadById(robotId);
+        var robot = await _robotService.ReadById(scheduleLocalizationMissionQuery.RobotId);
         if (robot == null)
         {
-            _logger.LogWarning("Could not find robot with id={id}", robotId);
+            _logger.LogWarning("Could not find robot with id={id}", scheduleLocalizationMissionQuery.RobotId);
             return NotFound("Robot not found");
         }
 
         if (robot.Status is not RobotStatus.Available)
         {
             _logger.LogWarning(
                 "Robot '{id}' is not available ({status})",
-                robotId,
+                scheduleLocalizationMissionQuery.RobotId,
                 robot.Status.ToString()
             );
             return Conflict($"The Robot is not available ({robot.Status})");
         }
+        var deck = await _assetDeckService.ReadById(scheduleLocalizationMissionQuery.DeckId);
+
+        if (deck == null)
+        {
+            _logger.LogWarning("Could not find deck with id={id}", scheduleLocalizationMissionQuery.DeckId);
+            return NotFound("Deck not found");
+        }
```

Comment:
This can be removed

Reply:
It is used below at robot.CurrentAssetDeck = deck

Reply:
Right, didn't see that one :)